test/radio/radio-controller-test.ts

```diff
@@ -24,6 +29,54 @@ class UdpServer {
   }
 }
 
+/**
+ * Class to manage a simple HTTP server for testing GET and POST
+ */
+class HttpServer {
+  private server: http.Server | undefined;
+  private onChunkWritten: (chunk: Buffer) => void = () => {};
+
+  /**
+   * Starts the HTTP server.
+   * Listens on two paths for GET and POST requests.
+   * @param listenPort  The port to start the server on.
+   */
+  public startServer(listenPort = 5000) {
+    this.server = http.createServer((request, response) => {
+      if (request.url === '/') {
+        response.writeHead(200);
+        response.end(HTTP_SERVER_BODY_1);
+      } else if (request.url === HTTP_SERVER_ENDPOINT_PATH) {
+        request.on('data', chunk => {
+          this.onChunkWritten(chunk);
```

Comment:
May want to adjust your server to also return multiple chunks of data to make sure that works.

Reply:
Done.